web/modules/custom/ai_screening_project/src/Helper/ProjectHelper.php

```diff
@@ -142,24 +142,28 @@ public function deleteRelated(NodeInterface $project): void {
         ->condition('project_id', $project->id(), '=')
         ->execute();
 
-      // Get all webforms that reference these project tracks.
-      $submissionIds = $this->webformSubmissionStorage->getQuery()
-        ->accessCheck(FALSE)
-        ->condition('entity_type', 'project_track', '=')
-        ->condition('entity_id', $projectTrackIds, 'IN')
-        ->execute();
-
-      // Delete project tracks and webform submissions for the project.
-      $projectTracks = $this->projectTrackStorage->loadMultiple($projectTrackIds);
-      foreach ($projectTracks as $projectTrack) {
-        $projectTrack->delete();
+      $submissionIds = [];
+      foreach ($projectTrackIds as $projectTrackId) {
+        // Get all webforms that reference these project tracks.
+        $submissionIds = $this->webformSubmissionStorage->getQuery()
+          ->accessCheck(FALSE)
+          ->condition('entity_type', 'project_track', '=')
+          ->condition('entity_id', $projectTrackId, '=')
+          ->execute();
       }
 
+      $projectTracks = $this->projectTrackStorage->loadMultiple($projectTrackIds);
       $webformSubmissions = $this->webformSubmissionStorage->loadMultiple($submissionIds);
+
+      // Delete project tracks and webform submissions for the project.
       foreach ($webformSubmissions as $webformSubmission) {
         $webformSubmission->delete();
       }
 
+      foreach ($projectTracks as $projectTrack) {
+        $projectTrack->delete();
+      }
+
     }
     catch (\Exception $exception) {
       $this->error('Error deleting project: @message', [
```

Comment:
We've recently added an `AbstractHelper` with a `logException` method  (https://github.com/itk-dev/ai-screening/blob/develop/web/modules/custom/ai_screening/src/Helper/AbstractHelper.php). Consider extending the abstract helper to reuse the exception logging (this will make it easier to send log messages to Prometheus in the future).